server/service/http/message/query/row.rs

```diff
@@ -48,7 +49,12 @@ pub fn encode_row<'a>(
         )?;
         encoded_row.insert(variable.as_str(), row_entry);
     }
-    let involved_blocks = row.provenance().branch_ids().map(|b| b.0 as u8).collect();
+    let involved_blocks = row
+        .provenance()
+        .branch_ids()
+        .map(|b| b.0)
+        .chain(always_taken_blocks.unwrap_or(&vec![]).iter().map(|b| b.0))
+        .collect();
```

Comment:
Append always taken blocks (root blocks of stages) to the BlockIDs from provenance.